sklearn/manifold/spectral_embedding_.py

```diff
@@ -116,6 +116,13 @@ def _set_diag(laplacian, value):
     return laplacian
 
 
+def eigen_sign_flip(u):
+    max_abs_rows = np.argmax(np.abs(u), axis = 1)
+    signs = np.sign(u[xrange(u.shape[0]), max_abs_rows])
```

Comment:
Or use this import `from ..externals.six.moves import xrange` here that would be fine...